skopt/tests/test_optimizer.py

```diff
@@ -194,3 +196,21 @@ def test_optimizer_base_estimator_string_smoke(base_estimator):
     opt = Optimizer([(-2.0, 2.0)], base_estimator=base_estimator,
                     n_initial_points=1, acq_func="EI")
     opt.run(func=lambda x: x[0]**2, n_iter=3)
+
+
+def test_defaults_are_equivalent():
```

Comment:
I think this is nice. Consider using a function that takes as input categorical, numerical and real values, in order to test consistency with a variety of dimension types, in order to possibly catch more bugs. For example:
```python
space = [('1', '2', '3'), (1, 3), (0.5, 1.5)]
def fnc(x):
    return float(x[0]) + float(x[1]) + float(x[2])
```